src/components/FilterBuilder/FilterBuilder.ts

```diff
@@ -0,0 +1,182 @@
+   *
+   * @param searchCatalogs - Array of catalog UUIDs from the search context.
+   * @param catalogUuidsToCatalogQueryUuids - Mapping from catalog UUID → query UUID.
+   * @returns The current FilterBuilder instance for chaining.
+   *
+   * @example
+   *   const catalogs = ['cat1', 'cat2'];
+   *   const mapping = { cat1: 'q1', cat2: 'q2' };
+   *   new FilterBuilder().filterByCatalogQueryUuids(catalogs, mapping).build()
+   *   // → "(enterprise_catalog_query_uuids:q1 OR enterprise_catalog_query_uuids:q2)"
+   */
+  filterByCatalogQueryUuids(
+    searchCatalogs: string[],
+    catalogUuidsToCatalogQueryUuids: Record<string, string>,
+  ): this {
+    const resolvedUuids = searchCatalogs
+      .map(catalog => catalogUuidsToCatalogQueryUuids[catalog])
+      .filter(Boolean);
+
+    return this.or('enterprise_catalog_query_uuids', resolvedUuids);
+  }
+
+  /**
+   * Conditionally excludes video content using the `FEATURE_ENABLE_VIDEO_CATALOG` flag.
+   * Adds the clause `NOT content_type:video` if the feature is disabled.
+   *
+   * @returns The current FilterBuilder instance for chaining.
+   *
+   * @example
+   *   new FilterBuilder().excludeVideoContentIfFeatureDisabled().build()
+   *   // → "NOT content_type:video" (if feature flag is off)
+   *   // → "" (if feature flag is on)
+   */
+  excludeVideoContentIfFeatureDisabled(): this {
+    if (!features.FEATURE_ENABLE_VIDEO_CATALOG) {
+      this.andRaw('NOT content_type:video');
+    }
+    return this;
+  }
+
+  /**
+   * Adds a filter for a single enterprise customer UUID.
+   *
+   * @param uuid - The UUID of the enterprise customer.
+   * @returns The current FilterBuilder instance for chaining.
+   *
+   * @example
+   *   new FilterBuilder().filterByEnterpriseCustomerUuid('abc-123').build()
+   *   // → "enterprise_customer_uuids:abc-123"
+   */
+  filterByEnterpriseCustomerUuid(uuid: string): this {
+    if (uuid) {
+      this.and('enterprise_customer_uuids', uuid);
+    }
+    return this;
+  }
+
+  /**
+   * Adds a filter for one or more catalog UUIDs.
+   *
+   * @param uuids - Array of catalog UUIDs to include.
+   * @returns The current FilterBuilder instance for chaining.
+   *
+   * @example
+   *   new FilterBuilder().filterByCatalogUuids(['c1', 'c2']).build()
+   *   // → "(enterprise_catalog_uuids:c1 OR enterprise_catalog_uuids:c2)"
+   */
+  filterByCatalogUuids(uuids: string[]): this {
+    return this.or('enterprise_catalog_uuids', uuids);
+  }
+
+  /**
+   * Builds and returns the final Algolia-compatible filter string.
+   *
+   * @returns A complete filter expression with `AND`-joined clauses.
+   *
+   * @example
+   *   new FilterBuilder()
+   *     .and('type', 'course')
+   *     .or('level', ['beginner', 'intermediate'])
+   *     .andRaw('NOT content_type:video')
+   *     .build()
+   *   // → "type:course AND (level:beginner OR level:intermediate) AND NOT content_type:video"
+   */
+  build(): string {
```

Comment:
nit: i believe you could rely on TS to infer the type from the return. TS knows the type of this.filters, and should be able to infer that `.join` results in a string without an explicit type hint.